PasswordValidator/Controllers/PasswordValidationController.cs

```diff
@@ -4,15 +4,28 @@ namespace PasswordValidator.Controllers
 {
     public class PasswordValidationController : Controller
     {
-        readonly SimpleValidatorFactory _SimpleValidatorFactory = new SimpleValidatorFactory();
-        readonly AdvancedValidatorFactory _AdvancedValidatorFactory = new AdvancedValidatorFactory();
+        readonly IValidatorFactory _validatorFactory;
+
+        public PasswordValidationController(IValidatorFactory validatorFactory)
+        {
+            _validatorFactory = validatorFactory;
+        }
 
         [HttpGet]
         [Route("ValidatePassword")]
         public ActionResult ValidatePassword(string password)
         {
-            IPasswordValidator simpleValidator = _SimpleValidatorFactory.CreateValidator();
-            bool isValid = simpleValidator.Validate(password);
+            bool isValid;
+
+            try
+            {
+                var simpleValidator = _validatorFactory.GetPasswordValidator("Simple");
+                isValid = simpleValidator.Validate(password);
+            }
+            catch (Exception)
+            {
+                return StatusCode(500);
+            }
 
             if (isValid)
```

Comment:
this if..else needs to be inside the `try` above

Reply:
Why?